mate-session/gsm-app.c

```diff
@@ -90,20 +131,36 @@ register_app (GsmApp *app)
 {
         GError *error;
         GsmAppPrivate *priv;
+        GsmExportedApp *skeleton;
 
         error = NULL;
         priv = gsm_app_get_instance_private (app);
 
-        priv->connection = dbus_g_bus_get (DBUS_BUS_SESSION, &error);
-        if (priv->connection == NULL) {
-                if (error != NULL) {
-                        g_critical ("error getting session bus: %s", error->message);
-                        g_error_free (error);
-                }
+        priv->connection = g_bus_get_sync (G_BUS_TYPE_SESSION, NULL, &error);
+        if (error != NULL) {
+                g_critical ("error getting session bus: %s", error->message);
+                g_error_free (error);
+                return FALSE;
+        }
+
+        skeleton = gsm_exported_app_skeleton_new ();
+        priv->skeleton = skeleton;
+        g_dbus_interface_skeleton_export (G_DBUS_INTERFACE_SKELETON (skeleton),
+                                          priv->connection, priv->id,
+                                          &error);
+
+        if (error != NULL) {
+                g_critical ("error registering app on session bus: %s", error->message);
+                g_error_free (error);
```

Comment:
You should clear `skeleton` here:
```c
g_clear_object(&skeleton)
```
Not sure about `priv->connection`... maybe needs to be cleared too?